src/memo/memo.ml

```diff
@@ -912,6 +944,14 @@ end = struct
                     ({ Exn_with_backtrace.exn; backtrace }, false)
                   | exn -> (exn, acc_reproducible)
                 in
+                let exn =
+                  match exn.exn with
```

Comment:
I would factor out this logic into the `Error` module. If the module is renamed it to `User_error` (as I suggested below), then the corresponding function could be called `User_error.capture exn` or something like this. I think the stack can be computed right that function without the need for `lazy`.

Reply:
Sounds good. I did that and removed the lazy, which was probably superfluous. Regarding capturing the call stack at this point, I wrote the code this way as when we reach this program point we already just captured the callstack. So it seemed better to me to reuse it rather than spread fiber up to here and re-capture the callstack. WDYT?

Reply:
Thanks! Yes, it seems fine to capture the stack here and pass it to `compute`.